### PR TITLE
Create new github repo url check

### PR DESCRIPTION
**Pull request by Josehower**

### CODE REVIEW

**Review by karlhorky — APPROVED** [review]

File: src/checks/linkOnGithubAbout.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import cheerio from 'cheerio';
+
+export const title = 'Github Repository contains link on About Section';
+
+export default async function linkOnGithubAbout() {
+  const { stdout: repoUrlStdout } = await execa.command(
+    'git remote get-url origin',
+  );
+
+  const html = await (await fetch(repoUrlStdout)).text();
```
Comment (karlhorky) [inline src/checks/linkOnGithubAbout.ts:12]: I guess here we should convert this to a URL, in case it's an SSH URL

Comment (karlhorky) [reply src/checks/linkOnGithubAbout.ts:12]: Maybe with `git-url-parse`:

https://github.com/IonicaBizau/git-url-parse

Comment (karlhorky) [reply src/checks/linkOnGithubAbout.ts:12]: Or maybe it's simple enough to do on our own:

```
'git@github.com:upleveled/preflight.git'
  .replace('git@github.com:', 'https://github.com/')
  .replace('.git', '')
```

File: src/checks/linkOnGithubAbout.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import execa from 'execa';
+import fetch from 'node-fetch';
+import cheerio from 'cheerio';
+
+export const title = 'Github Repository contains link on About Section';
```
Comment (karlhorky) [inline src/checks/linkOnGithubAbout.ts:5]: GitHub has a capital `H`

File: src/checks/linkOnGithubAbout.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      return (
+        $(this)
+          .text()
+          .trim() === 'About'
+      );
+    })
+    .next()
+    .next()
+    .text()
+    .trim();
```
Comment (karlhorky) [inline src/checks/linkOnGithubAbout.ts:29]: This technique is:

- too strict in one way: needs to match that regex
- too lax in another way: only needs to find a URL **somewhere** in the text

I would suggest trying to match only existence of the `.octicon.octicon-link` and the `a[href]` elements - this will be enough to confirm that they've added some link.

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,4 @@
 *.log
 .DS_Store
-node_modules
+node_modules/
```
Comment (karlhorky) [inline .gitignore:3]: Why did you add this? If it's causing Preflight to fail without it, we should fix this in Preflight...

Comment (karlhorky) [reply .gitignore:3]: I've removed the slash again in `main`. For me, Preflight runs through.

Comment (karlhorky) [reply .gitignore:3]: If it doesn't for you (on Windows), then let's fix the issue in Preflight, not change the `.gitignore`.

File: package.json [hunk 5]
```diff
@@ -59,9 +59,13 @@
     "typescript": "4.1.3"
   },
   "dependencies": {
+    "@types/cheerio": "0.22.23",
+    "@types/node-fetch": "2.5.8",
```
Comment (karlhorky) [inline package.json:63]: Types should always be in `devDependencies`

File: src/index.ts [hunk 6]
```diff
@@ -22,6 +23,7 @@ const listrTasks = [
   nodeModulesIgnoredFromGit,
   noExtraneousFilesCommittedToGit,
   noSecretsCommittedToGit,
+  linkOnGithubAbout,
```
Comment (karlhorky) [inline src/index.ts:26]: This is an asynchronous check - should be in the "Async" block below.

File: .gitignore (hunk 4 above)
Comment (Josehower) [reply .gitignore:3]: i think was failin for other reason an di thought was this, i changed to test and i forgot to change it back

File: src/index.ts (hunk 6 above)
Comment (Josehower) [reply src/index.ts:26]: sorry, for that i didnt read i just added to the list. now i know
